CIP-0001 | Annual overhaul and process update (#924)
cardano-foundation/CIPs: CIP-0001/README.md

    - Matthias Benkort <matthias.benkort@cardanofoundation.org>
    - Duncan Coutts <duncan.coutts@iohk.io>
    - Michael Peyton Jones <michael.peyton-jones@iohk.io>
    - Robert Phair <rphair@cosd.com>
Implementors: N/A
Discussions:
    - https://github.com/cardano-foundation/cips/pull/366
    - https://github.com/cardano-foundation/cips/pull/331
    - https://github.com/cardano-foundation/CIPs/tree/3da306f3bfe89fa7de8fe1bf7a436682aeee25c5/CIP-0001#abstract
    - https://github.com/cardano-foundation/CIPs/pull/924
Created: 2020-03-21
License: CC-BY-4.0
---

## Abstract

> A Cardano Improvement Proposal (CIP) is a formalised design document for the Cardano community and the name of the process by which such documents are produced and listed. A CIP provides information or describes a change to the Cardano ecosystem, processes, or environment concisely and in sufficient technical detail. In this CIP, we explain what a CIP is; how the CIP process functions; the role of the CIP Editors; and how users should go about proposing, discussing and structuring a CIP.

Reviewer: ```suggestion
A Cardano Improvement Proposal (CIP) is a formalised design document for the Cardano community and the name of the process by which such documents are produced and listed. A CIP provides information or describes a change to the Cardano ecosystem, processes, or environment concisely and in sufficient technical detail. In this CIP, we explain what a CIP is, how the CIP process functions, the role of the CIP Editors, and how users should go about proposing, discussing, and structuring a CIP.
```
some grammar, I feel this works better

Author: Both are grammatically correct (since the last noun phrase contains commas) but I agree with you since it avoids people stopping for a "breath" when passing the semicolon 😅

    - Duncan Coutts <duncan.coutts@iohk.io>
Implementors: N/A
Discussions:

>     - https://github.com/cardano-foundation/cips/pull/366

Reviewer: ```suggestion
    - https://github.com/cardano-foundation/CIPs/pull/366
```
capitalisation, it doesnt really matter of course

Author: good spot... I'll fix the non-canonical names in the header as well.

Created: 2020-03-21
License: CC-BY-4.0
---
```

Especially because Markdown link syntax is not supported in the header preamble, labels can be added to clarify list items; e.g.:

> ```

Reviewer: ```suggestion
```yaml
```

Author: oops 🤓

> Implementors:

Reviewer: ```suggestion
Discussions:
 - Original PR: https://github.com/cardano-foundation/CIPs/pull/366
```
I feel like this works as a better example, as `Implementors` can be referenced by their website address

Author: I agree with you, but wouldn't the YAML format label need double quotes around it? (https://docs.nobl9.com/yaml-guide)  If so, does this look OK?  And it would it render with or without quotes in the GitHub header?

CIPs are classified into distinct categories that help organise (and thus, find) them. Categories are meant to be flexible and evolve as new domains emerge. Authors may leave the category as `?` should they not be sure under which category their proposal falls; editors will eventually assign one or reject the proposal altogether should it relate to an area where the CIP process does not apply.

Submissions can be made to these categories whenever relevant, without following any particular submission guidelines:

Category | Description
---      | ---
Meta     | Designates meta-CIPs, such as this one, which typically serve another category or group of categories
Wallets  | For standardisation across wallets (hardware, full-node or light)
Tokens   | About tokens (fungible or non-fungible) and minting policies in general
Metadata | For proposals around metadata (on-chain or off-chain)
Tools    | A broad category for ecosystem features not falling into any other category

> Additionally, representatives of ecosystem categories may explicitly _enlist_ their categories (see next section) to suggest a closer relationship with the CIP process.  The following categories are confirmed as enslisted according to CIPs which define that relationship:

Reviewer: ```suggestion
Additionally, representatives of ecosystem categories may explicitly _enlist_ their categories (see next section) to suggest a closer relationship with the CIP process. The following categories are confirmed as enlisted according to CIPs which define that relationship:
```
spelling and extra space

Author: N.B. the "extra" space is not a problem here: https://github.com/cardano-foundation/CIPs/pull/893#discussion_r1767656606

- defining boundaries within their project for which the CIP process does apply;
- establishing points of contact and any designated reviews for a category;
- agreeing upon how proposals move from the state of idea (i.e. CIP) to actual implementation work;
- writing CIPs for significant changes introduced in their projects when it applies.

Any guidelines for this cooperation should be described by a dedicated CIP whenever possible.  When such a CIP is posted or supersedes another one, it will be entered into the above table in the Categories section.  Participants of enlisted categories should follow the requirements outlined in that CIP and should update such proposals whenever these requirements or relationships change.

> **Warning** A positive review by any enlisted project representative does not constitute a commitment to implement the CIP. It is still the CIP author's responsibility to create an implementation plan and identify implementors.

Editors occasionally invite representatives from enlisted categories to speak during review meetings and solicit them for ultimate approvals of proposals in their area of expertise.

> > **Note** Optionally, projects may show their enlisting using the following badge on their introductory README: ![](https://github.com/cardano-foundation/CIPs](https://raw.githubusercontent.com/cardano-foundation/CIPs/master/.github/badge.svg)

Reviewer: ```suggestion
**Note** Optionally, projects may show their enlisting using the following badge on their introductory README: ![](https://github.com/cardano-foundation/CIPs](https://raw.githubusercontent.com/cardano-foundation/CIPs/master/.github/badge.svg)
```

This part doesnt render properly, maybe we should take out the `>`

Author: sure; let's try it.  That code is an eyesore everywhere and leaves us looking over our shoulder on derived sites like `cips.cardano.org` and the Dev Portal.  It does look dubious as written but I'd never suggested a correction since I didn't really know how the embed was supposed to look.  If still broken anywhere after the merge, we can move it to a Gist for further testing, with any project that might be using this badge (?), and then fix it in another PR.

Tools    | A broad category for ecosystem features not falling into any other category

Additionally, representatives of ecosystem categories may explicitly _enlist_ their categories (see next section) to suggest a closer relationship with the CIP process. The following categories are confirmed as enlisted according to CIPs which define that relationship:

Category | Description
---      | ---
Plutus   | Changes or additions to Plutus, following the process described in [CIP-0035][]
Ledger   | For proposals regarding the Cardano ledger, following the process described in [CIP-0084][]

These tentatively enlisted categories await CIPs to describe any enlistment relationship:

Category  | Description
---       | ---
Catalyst  | For proposals affecting Project Catalyst or the Jörmungandr project
Consensus | For proposals affecting implementations of the Cardano Consensus layer and algorithms
Network   | Specifications and implementations of Cardano's network protocols and applications

#### Project Enlisting

Project representatives intending to follow an "enlisted" category above agree to coordinate with related development by sharing efforts to review and validate new proposals.

> It should be noted that single organisations can no longer repesent any ecosystem or development category, which makes these enlistment guidelines both decentralised and cooperative, including whenver possible:

Reviewer 2: ```suggestion
It should be noted that single organisations can no longer represent any ecosystem or development category, which makes these enlistment guidelines both decentralised and cooperative, including whenever possible:
```

[@Crypto2099]: https://github.com/Crypto2099
[@perturbing]: https://github.com/perturbing

> Emeritus editors:

Reviewer: Very nice idea, love it!

Author: thanks & I believe it's also a way to pre-empt future issues like the one that developed here:
- https://github.com/cardano-foundation/CIPs/pull/366#pullrequestreview-1170652027

... for which I had suggested this as a solution to the deadlock we had over it: https://github.com/cardano-foundation/CIPs/pull/366#issuecomment-1308482496
